wagtail_streamfield_migration_toolkit/operations.py

```diff
@@ -208,8 +206,11 @@ def apply(self, block_value):
             else:
                 mapped_block_value.append(child_block)
 
-        new_stream_block = {"type": self.stream_block_name, "value": stream_value}
-        mapped_block_value.append(new_stream_block)
+        # Append the new stream block only if it's not empty
+        if len(stream_value) > 0:
```

Comment:
```suggestion
        if stream_value:
```

Just a suggestion, what you have is good - we don't really care about the exact length of `stream_value`, so checking that it's truthy (non-empty) is enough in this case and doesn't introduce the function call overhead